packages/playwright-test/src/dispatcher.ts

```diff
@@ -494,9 +495,15 @@ class Worker extends EventEmitter {
       const { method, params } = message;
       this.emit(method, params);
     });
+
+    this.ready = new Promise((resolve, reject) => {
+      this.process.once('exit', () => reject(new Error('worker exited before it became ready')));
```

Comment:
This assumes that failure to create a worker is considered a non-recoverable error and we crash the entirety of the test runner upon it. @dgozman, would we want to handle it and retry?

Reply:
Yep, we should handle this case without crashing 😄 

I propose the following changes:
- make `_ready` resolve with a boolean (success or failed);
- make `init` return immediately if `_ready` failed, so that it does not stall.
- make `_runJob` check `worker.didExit`, and [call `onExit`](https://github.com/microsoft/playwright/blob/3636d8548f5ad7740fe9db53cb0bca2e0de649c0/packages/playwright-test/src/dispatcher.ts#L393) immediately if the worker did exit already. This will mark scheduled tests as failed, and enqueue a retry if needed.

Reply:
Scratch that, let's crash instead, since something is very much broken in this case.